theme/static/js/hs-vue/geoconnex.js

```diff
@@ -1246,23 +1241,49 @@ const geoconnexApp = new Vue({
         geoconnexApp.appMessages.push({ message: message, level: level });
       }
     },
-    setFeatureName(feature) {
+    async setFeatureName(feature) {
       const geoconnexApp = this;
-      const nameField = geoconnexApp.getFeatureNameField(feature.collection);
+      const nameField = await geoconnexApp.getFeatureNameField(
+        feature.collection
+      );
       feature.NAME = feature.properties[nameField] || "";
     },
-    getFeatureNameField(collectionName) {
+    async getFeatureNameField(collectionName) {
       const geoconnexApp = this;
-      // This could also be accomplished by flattening json-ld for the feature and searching for the "https://schema.org/name"
-      return geoconnexApp.featureNameFieldMap[collectionName] || "NAME";
+      const url = `${geoconnexApp.geoconnexUrl}/${collectionName}/items?f=jsonld&lang=en-US&skipGeometry=true&limit=1`;
+      const featureJsonLd = await geoconnexApp.fetchURLFromCacheOrGeoconnex({
+        url: url,
+      });
+      let array = featureJsonLd["@context"];
```

Comment:
This array can be declared as a constant as well. And a better name could be used.